fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt

```diff
@@ -137,6 +147,39 @@ class DefaultReviewQualityCheckService(
                 }
             }
         }
+
+    override suspend fun recordRecommendedProductClick(productAid: String) =
+        withContext(Dispatchers.Main) {
+            suspendCoroutine { continuation ->
+                browserStore.state.selectedTab?.engineState?.engineSession?.sendClickAttributionEvent(
```

Comment:
I can't imagine doing an optional call chain like this in the Java days 😅

Reply:
_nullpointerexceptions intensifies_ 😅 

to be fair, we could theoretically assert these as non null since these calls can only happen from UI actions when we have a selected tab, but it's safer to not assume that since this class doesn't know about where the triggers can come from.